### ISSUE TITLE
audit loging - several loginattempts

### ISSUE THREAD

**Issue by aignerat** [nextcloud/android#7008]
### Steps to reproduce
1. Login Nextcloud on android app

### Expected behaviour
- audit.log triggers 1 login attempt and 1 login successful log entry on successful login

### Actual behaviour
- several entries are made, there are 2 login attempts and 1 login successful triggered 6 times on entering the app

### Can you reproduce this problem on https://try.nextcloud.com?
not possible, no access to logs

### Environment data
Android version:
10.0.0.232

Device model: 
Huawai Travel Mate 20 lite

Stock or customized system:

Nextcloud app version:
3.13.1 
on older versions there is the same problem with 2 login attempts and 1 login successful in the logs, but not several times on every login

Nextcloud server version:
v15.0.4 + v19.0.1
Reverse proxy:

#### audit.log
```
{"reqId":"X22qR-39yw-f63g0kR6ihAAAAEA","level":1,"time":"2020-09-25T10:28:56+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"GET","url":"/index.php/avatar/pseudoname/384","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qR-39yw-f63g0kR6ihAAAAEA","level":1,"time":"2020-09-25T10:28:56+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"GET","url":"/index.php/avatar/pseudoname/384","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qR-39yw-f63g0kR6ihAAAAEA","level":1,"time":"2020-09-25T10:28:56+02:00","remoteAddr":"192.168.123.234","user":"pseudoname","app":"admin_audit","method":"GET","url":"/index.php/avatar/pseudoname/384","message":"Login successful: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qR-39yw-f63g0kR6igwAAAEY","level":1,"time":"2020-09-25T10:28:56+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"GET","url":"/ocs/v1.php/cloud/user?format=json","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qR-39yw-f63g0kR6igwAAAEY","level":1,"time":"2020-09-25T10:28:56+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"GET","url":"/ocs/v1.php/cloud/user?format=json","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qR-39yw-f63g0kR6igwAAAEY","level":1,"time":"2020-09-25T10:28:56+02:00","remoteAddr":"192.168.123.234","user":"pseudoname","app":"admin_audit","method":"GET","url":"/ocs/v1.php/cloud/user?format=json","message":"Login successful: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qSIeHCt9MMZ46-ZqBBwAAABE","level":1,"time":"2020-09-25T10:28:56+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"GET","url":"/ocs/v2.php/cloud/capabilities?format=json","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qSIeHCt9MMZ46-ZqBBwAAABE","level":1,"time":"2020-09-25T10:28:56+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"GET","url":"/ocs/v2.php/cloud/capabilities?format=json","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qSIeHCt9MMZ46-ZqBBwAAABE","level":1,"time":"2020-09-25T10:28:56+02:00","remoteAddr":"192.168.123.234","user":"pseudoname","app":"admin_audit","method":"GET","url":"/ocs/v2.php/cloud/capabilities?format=json","message":"Login successful: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qSP39yw-f63g0kR6ihQAAAEc","level":1,"time":"2020-09-25T10:28:57+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"POST","url":"/ocs/v2.php/apps/notifications/api/v2/push?format=json","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qSP39yw-f63g0kR6ihQAAAEc","level":1,"time":"2020-09-25T10:28:57+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"POST","url":"/ocs/v2.php/apps/notifications/api/v2/push?format=json","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qSP39yw-f63g0kR6ihQAAAEc","level":1,"time":"2020-09-25T10:28:57+02:00","remoteAddr":"192.168.123.234","user":"pseudoname","app":"admin_audit","method":"POST","url":"/ocs/v2.php/apps/notifications/api/v2/push?format=json","message":"Login successful: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qSYeHCt9MMZ46-ZqBCQAAAA8","level":1,"time":"2020-09-25T10:28:57+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"GET","url":"/ocs/v2.php/apps/files/api/v1/directEditing?format=json","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qSYeHCt9MMZ46-ZqBCQAAAA8","level":1,"time":"2020-09-25T10:28:57+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"GET","url":"/ocs/v2.php/apps/files/api/v1/directEditing?format=json","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qSYeHCt9MMZ46-ZqBCQAAAA8","level":1,"time":"2020-09-25T10:28:57+02:00","remoteAddr":"192.168.123.234","user":"pseudoname","app":"admin_audit","method":"GET","url":"/ocs/v2.php/apps/files/api/v1/directEditing?format=json","message":"Login successful: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qS4eHCt9MMZ46-ZqBFwAAABM","level":1,"time":"2020-09-25T10:28:59+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"GET","url":"/index.php/apps/files/api/v1/thumbnail/256/256/Nextcloud.png","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qS4eHCt9MMZ46-ZqBFwAAABM","level":1,"time":"2020-09-25T10:28:59+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"GET","url":"/index.php/apps/files/api/v1/thumbnail/256/256/Nextcloud.png","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
{"reqId":"X22qS4eHCt9MMZ46-ZqBFwAAABM","level":1,"time":"2020-09-25T10:28:59+02:00","remoteAddr":"192.168.123.234","user":"pseudoname","app":"admin_audit","method":"GET","url":"/index.php/apps/files/api/v1/thumbnail/256/256/Nextcloud.png","message":"Login successful: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
@nextcloud/server-triage 
Client always sends auth data. Do you have an idea why not everywhere username is shown?

> {"reqId":"X22qR-39yw-f63g0kR6igwAAAEY","level":1,"time":"2020-09-25T10:28:56+02:00","remoteAddr":"192.168.123.234","user":"--","app":"admin_audit","method":"GET","url":"/ocs/v1.php/cloud/user?format=json","message":"Login attempt: \"pseudoname\"","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"19.0.1.1"}


Should at least be with logged in user…

**Comment by aignerat** [comment]
the name is at least in the params, webinterface does it the same way for audit.log. I didn't look into this matter as I write the auditlog into the database with a reporting-app where I use params['uid'] as username. I tried the same procedure with an old nextcloud-android-app, I got 2 loginattempts and 1 login successful too. The iOS-App has the same issue with 2 loginattempt entries and 1 successful login afterwards. I guess the log takes the username of the session, of course there is no username involved when you're not logged in.
